File: leetsync/3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.go
func areaOfMaxDiagonal(dimensions [][]int) int {
	n := len(dimensions)
	maxDiagonal, maxArea := 0, 0
	for i := 0; i < n; i++ {
		length := dimensions[i][0]
		width := dimensions[i][1]
		diagonal := length*length + width*width
		area := length * width
		if maxDiagonal < diagonal {
			maxDiagonal = diagonal
			maxArea = area
		} else if maxDiagonal == diagonal {
			maxArea = max(maxArea, area)
		}
	}
	return maxArea
}

File: leetsync/3251-maximum-area-of-longest-diagonal-rectangle/maximum-area-of-longest-diagonal-rectangle.java
class Solution {
    public int areaOfMaxDiagonal(int[][] dimensions) {
        int n = dimensions.length;
        int maxDiagonal = 0, maxArea = 0;
        for (int i = 0; i < n; i++) {
            int length = dimensions[i][0];
            int width = dimensions[i][1];
            int diagonal = length * length + width * width;
            int area = length * width;
            if (maxDiagonal < diagonal) {
                maxDiagonal = diagonal;
                maxArea = area;
            } else if (maxDiagonal == diagonal) {
                maxArea = Math.max(maxArea, area);
            }
        }
        return maxArea;
    }
}